File: CookieClickerJava/app/src/main/java/com/example/cookieclickerjava/PrestigeShopActivity.java
package com.example.cookieclickerjava;

import android.content.SharedPreferences;
import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.example.cookieclickerjava.databinding.ActivityPrestigeShopBinding;

import java.util.ArrayList;
import java.util.List;

public class PrestigeShopActivity extends AppCompatActivity {

    private ActivityPrestigeShopBinding b;
    private SharedPreferences prefs;

    private int prestigePoints = 0;
    private List<PrestigePerk> perks; // sınıf seviyesi

    private Runnable onAdapterRefresh;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        b = ActivityPrestigeShopBinding.inflate(getLayoutInflater());
        setContentView(b.getRoot());

        if (getSupportActionBar() != null) {
            getSupportActionBar().setTitle(getString(R.string.shop_title));
        }

        prefs = getSharedPreferences("cookie_prefs", MODE_PRIVATE);

        // State yükle
        prestigePoints = prefs.getInt("prestige_points", 0);

        // Context hazır, getString kullanılabilir
        perks = new ArrayList<>();
        perks.add(new PrestigePerk(
                "perk_gprod",
                getString(R.string.perk_gprod),
                getString(R.string.perk_gprod_desc, 5),
                1,
                1.6,
                0,
                Integer.MAX_VALUE
        ));
        perks.add(new PrestigePerk(
                "perk_crit",
                getString(R.string.perk_crit),
                getString(R.string.perk_crit_desc, 1, 3),
                2,
                1.7,
                0,
                Integer.MAX_VALUE
        ));
        perks.add(new PrestigePerk(
                "perk_discount",
                getString(R.string.perk_discount),
                getString(R.string.perk_discount_desc, 2, 50),
                3,
                1.8,
                0,
                25 // %50 tavan
        ));
        perks.add(new PrestigePerk(
                "perk_taptop",
                getString(R.string.perk_taptop),
                getString(R.string.perk_taptop_desc),
                2,
                1.5,
                0,
                Integer.MAX_VALUE
        ));

        // Kayıtlı seviyeleri yükle
        for (PrestigePerk perk : perks) {
            perk.setLevel(prefs.getInt(perk.getKey(), 0));
        }

        PrestigeShopAdapter adapter = new PrestigeShopAdapter(
                perks,
                cost -> prestigePoints >= cost,
                perk -> {
                    buyPerk(perk);
                }
        );

        b.rvPerks.setLayoutManager(new LinearLayoutManager(this));
        b.rvPerks.setAdapter(adapter);

        updateHeader();

        onAdapterRefresh = () -> {
            adapter.notifyDataSetChanged();
            updateHeader();
        };
    }

    private void updateHeader() {
        b.tvPoints.setText(getString(R.string.prestige_points, prestigePoints));
    }

    private void buyPerk(PrestigePerk perk) {
        int cost = perk.costForNext();
        if (prestigePoints < cost || perk.getLevel() >= perk.getMaxLevel()) return;

        prestigePoints -= cost;
        perk.setLevel(perk.getLevel() + 1);

        prefs.edit()
                .putInt("prestige_points", prestigePoints)
                .putInt(perk.getKey(), perk.getLevel())
                .apply();

        if (onAdapterRefresh != null) onAdapterRefresh.run();
    }
}

File: CookieClickerKotlin/app/src/main/java/com/example/cookieclickerkotlin/PrestigeShopActivity.kt
package com.example.cookieclickerkotlin

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.cookieclickerkotlin.databinding.ActivityPrestigeShopBinding

class PrestigeShopActivity : AppCompatActivity() {

    private lateinit var b: ActivityPrestigeShopBinding
    private val prefs by lazy { getSharedPreferences("cookie_prefs", MODE_PRIVATE) }

    private var prestigePoints = 0
    private lateinit var perks: MutableList<PrestigePerk>   // <-- sınıf seviyesinde tanımla, içeride doldur

    private var onAdapterRefresh: (() -> Unit)? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        b = ActivityPrestigeShopBinding.inflate(layoutInflater)
        setContentView(b.root)
        supportActionBar?.title = getString(R.string.shop_title)

        // State yükle
        prestigePoints = prefs.getInt("prestige_points", 0)

        // !!! getString kullanımı artık burada (Context hazır)
        perks = mutableListOf(
            PrestigePerk(
                key = "perk_gprod",
                title = getString(R.string.perk_gprod),
                // seviye başına %5
                desc  = getString(R.string.perk_gprod_desc, 5),
                baseCost = 1,
                costScaling = 1.6
            ),
            PrestigePerk(
                key = "perk_crit",
                title = getString(R.string.perk_crit),
                // seviye başına %1 şans, x3 çarpan
                desc  = getString(R.string.perk_crit_desc, 1, 3),
                baseCost = 2,
                costScaling = 1.7
            ),
            PrestigePerk(
                key = "perk_discount",
                title = getString(R.string.perk_discount),
                // seviye başına %2 indirim, tavan %50
                desc  = getString(R.string.perk_discount_desc, 2, 50),
                baseCost = 3,
                costScaling = 1.8,
                maxLevel = 25 // %50 tavan
            ),
            PrestigePerk(
                key = "perk_taptop",
                title = getString(R.string.perk_taptop),
                desc  = getString(R.string.perk_taptop_desc),
                baseCost = 2,
                costScaling = 1.5
            )
        )


        // Kayıtlı seviyeleri yükle
        perks.forEach { it.level = prefs.getInt(it.key, 0) }

        val adapter = PrestigeShopAdapter(
            items = perks,
            canAfford = { cost -> prestigePoints >= cost },
            onBuy = { perk -> buyPerk(perk) }
        )
        b.rvPerks.layoutManager = LinearLayoutManager(this)
        b.rvPerks.adapter = adapter

        updateHeader()

        fun refresh() { adapter.notifyDataSetChanged(); updateHeader() }
        onAdapterRefresh = ::refresh
    }

    private fun updateHeader() {
        b.tvPoints.text = getString(R.string.prestige_points, prestigePoints)
    }

    private fun buyPerk(perk: PrestigePerk) {
        val cost = perk.costForNext()
        if (prestigePoints < cost || perk.level >= perk.maxLevel) return
        prestigePoints -= cost
        perk.level += 1
        prefs.edit()
            .putInt("prestige_points", prestigePoints)
            .putInt(perk.key, perk.level)
            .apply()
        onAdapterRefresh?.invoke()
    }
}
